### PR TITLE
Fix up parsing of OGR GML with ol.format.GML

### LINKED ISSUE #3098: assertion errors when reading OGR GML with ol.format.GML
Example GML:

```
<?xml version="1.0" encoding="utf-8" ?>
<wfs:FeatureCollection
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	 xmlns:wfs="http://www.opengis.net/wfs"
     xsi:schemaLocation="http://ogr.maptools.org/ test.xsd"
     xmlns:ogr="http://ogr.maptools.org/"
     xmlns:gml="http://www.opengis.net/gml">

                                                                                                                       
  <gml:featureMember>
    <ogr:Plaatsbepalingspunt fid="Plaatsbepalingspunt.0">
		<ogr:geometryProperty>
			<gml:Point>
				<gml:pos srsDimension="2">115512.666 479836.28</gml:pos>
			</gml:Point>
		</ogr:geometryProperty>
		<ogr:gml_id>x2</ogr:gml_id>
		<ogr:namespace>NL.IMGEO</ogr:namespace>
		<ogr:lokaalID>L0001.A3C177B4105A4FFD82EB80084C8CA732</ogr:lokaalID>
		<ogr:nauwkeurigheid>60</ogr:nauwkeurigheid>
		<ogr:datumInwinning>2014-02-14</ogr:datumInwinning>
		<ogr:inwinnendeInstantie>L0001</ogr:inwinnendeInstantie>
		<ogr:inwinningsmethode>fotogrammetrisch</ogr:inwinningsmethode>
    </ogr:Plaatsbepalingspunt>
  </gml:featureMember>
</wfs:FeatureCollection>
```

If featureMembers instead of featureMember parsing is okay. Also using ol.format.WFS parsing is okay. Will investigate deeper.

### PR DESCRIPTION
**Pull request by bartvde**
fixes #3098

### CODE REVIEW

File: src/ol/format/gml/gmlbaseformat.js [hunk 1]
```diff
@@ -99,6 +99,11 @@ ol.format.GMLBase.prototype.readFeatures_ = function(node, objectStack) {
     }
     var parsers = {};
     var parsersNS = {};
+    if (goog.object.get(context, 'useArrayPusher') === true) {
+      var ns = 'http://www.opengis.net/gml';
```
Comment (elemoine) [inline src/ol/format/gml/gmlbaseformat.js]: Shouldn't this be set as a constant under the `ol.format.GMLBase` namespace?

Comment (bartvde) [inline src/ol/format/gml/gmlbaseformat.js]: Yes this would be better, it's used in other places as well.

**Comment by bartvde** [comment]
@elemoine I updated the PR with the const

File: src/ol/format/gml/gmlbaseformat.js [hunk 2]
```diff
@@ -99,6 +106,11 @@ ol.format.GMLBase.prototype.readFeatures_ = function(node, objectStack) {
     }
     var parsers = {};
     var parsersNS = {};
+    if (goog.object.get(context, 'useArrayPusher') === true) {
+      goog.object.set(this.FEATURE_COLLECTION_PARSERS[ol.format.GMLBase.GMLNS],
+          'featureMember',
+          ol.xml.makeArrayPusher(ol.format.GMLBase.prototype.readFeatures_));
```
Comment (elemoine) [inline src/ol/format/gml/gmlbaseformat.js]: Do you really want to change the object on the prototype here?

Comment (bartvde) [inline src/ol/format/gml/gmlbaseformat.js]: I am not changing it on the prototype but on the instance (this.FEATURE_COLLECTION_PARSERS)

Comment (bartvde) [inline src/ol/format/gml/gmlbaseformat.js]: This needs to be changed to bracket notation given #3101

Comment (elemoine) [inline src/ol/format/gml/gmlbaseformat.js]: > I am not changing it on the prototype but on the instance (this.FEATURE_COLLECTION_PARSERS)

I think `this.FEATURE_COLLECTION_PARSERS` refers to the object [defined](https://github.com/bartvde/ol3/blob/ogr-gml2/src/ol/format/gml/gmlbaseformat.js#L130-L137) on the prototype.

**Comment by bartvde** [comment]
@elemoine I updated this PR to use a different approach, i.e. to change the readers on the instance (gmlFormat of ol.format.WFS) instead.

File: src/ol/format/wfsformat.js [hunk 3]
```diff
@@ -125,6 +125,9 @@ ol.format.WFS.prototype.readFeaturesFromNode = function(node, opt_options) {
   goog.object.extend(context, this.getReadOptions(node,
       goog.isDef(opt_options) ? opt_options : {}));
   var objectStack = [context];
+  this.gmlFormat_.FEATURE_COLLECTION_PARSERS[ol.format.GMLBase.GMLNS][
+      'featureMember'] =
+      ol.xml.makeArrayPusher(ol.format.GMLBase.prototype.readFeaturesInternal);
```
Comment (elemoine) [inline src/ol/format/wfsformat.js]: It seems to me that, again, you're changing the `FEATURE_COLLECTION_PARSERS` object here, which is set on the `prototype` and therefore shared by all the GML formats one might have.

Is it intended?

Comment (bartvde) [inline src/ol/format/wfsformat.js]: Ah you're right again :-) will change this, since it's not intended

Comment (bartvde) [inline src/ol/format/wfsformat.js]: @elemoine does 5d70e617539982b3ca10bea2f0da7dfbbfe11fed work for you or you had something else in mind here?

File: src/ol/format/gml/gmlbaseformat.js [hunk 4]
```diff
@@ -68,18 +68,36 @@ ol.format.GMLBase = function(opt_options) {
    */
   this.schemaLocation = '';
 
+  /**
+   * @type {Object.<string, Object.<string, Object>>}
+   */
+  this.FEATURE_COLLECTION_PARSERS = Object({
+    'http://www.opengis.net/gml': {
```
Comment (fredj) [inline src/ol/format/gml/gmlbaseformat.js]: `ol.format.GMLBase.GMLNS` should be used here

Comment (elemoine) [inline src/ol/format/gml/gmlbaseformat.js]: And the use of the `Object` constructor is not necessary anymore. It was there to fool `gjslint`.

**Comment by bartvde** [comment]
@fredj @elemoine PR updated

**Comment by elemoine** [comment]
LGTM.

**Comment by bartvde** [comment]
thanks for the reviews and the patience on this one :-)

**Pull request merged**